tyr/servers/nginx/server.py

```diff
@@ -32,8 +38,28 @@ def __init__(self, group=None, server_type=None, instance_type=None,
                                           environment, ami, region, role,
                                           keypair, availability_zone,
                                           security_groups, block_devices,
-                                          chef_path, subnet_id, dns_zones)
+                                          chef_path, subnet_id, dns_zones,
+                                          platform, use_latest_ami,
+                                          ingress_groups_to_add, ports_to_authorize,
+                                          classic_link, add_route53_dns)
 
     def configure(self):
-
         super(NginxServer, self).configure()
+
+    @property
+    def user_data(self):
+        # read in userdata file
+        user_data = None
+        try:
+            f = data_file('user_data_chef_provision')
+            user_data = f.read()
+
+            validation_key_path = os.path.expanduser('~/.chef/chef-validator.pem')
+            validation_key_file = open(validation_key_path, 'r')
+            validation_key = validation_key_file.read()
+
+            return user_data.format(validation_key=validation_key)
+
+        except IOError:
+            # Handle error reading file
+            pass
```

Comment:
This exception needs to be better handled; the user should be provided with some kind of information about the problem.